parserc.go

```diff
@@ -890,6 +893,24 @@ func yaml_parser_parse_block_mapping_value(parser *yaml_parser_t, event *yaml_ev
 	if token.typ == yaml_VALUE_TOKEN {
 		mark := token.end_mark
 		skip_token(parser)
+		// Move foot comment to head comment. This prevents that in some cases, comments
+		// in maps are moved to the wrong place. In the following map:
+		//
+		//     a:
+		//       b:
+		//         # comment followed by newline
+		//
+		//         c: d
+		//
+		// (the newline between the comment and the next line is essential!)
```

Comment:
This is stated more succinctly in the body of the yaml right above, so no need to repeat here.

Otherwise, thanks for the comment. It provides clear rationale for the non-obvious logic.